src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -115,6 +117,17 @@ async def exec_aggregation_async(
         result = self._process_http_response(response)
         return AggregationResult.deserialize(result)
 
+    async def ensure_executor_startup(self, error_file):
+        """Ensure the executor service is initialized before calling the API to get the results"""
+        try:
+            await self.ensure_executor_health()
```

Comment:
By the time we invoke health api? Does the executor server's validation logic finish? 
If not, then what is the response of the health api? If 200, then it might be a  problem because we might miss checking the error file.

Reply:
Only after validation is completed, the executor can be started. Before that, the health API cannot be called.